a_test.go

```diff
@@ -183,42 +185,374 @@ func TestA_uses_Logger_dynamic_interface(t *testing.T) {
 	}
 }
 
+func TestA_WithContext(t *testing.T) {
+	t.Parallel()
+	type ctxKeyT struct{}
+	var ctxKey ctxKeyT
+
+	for _, parallel := range []bool{false, true} {
+		parallel := parallel
+		for _, throwPanic := range []bool{false, true} {
+			throwPanic := throwPanic
+			t.Run("Parallel="+strconv.FormatBool(parallel)+" and panic="+strconv.FormatBool(throwPanic), func(t *testing.T) {
+				t.Parallel()
+
+				flow := &goyek.Flow{}
+
+				flow.SetOutput(io.Discard)
+				loggerSpy := &helperLoggerSpy{}
+				flow.SetLogger(loggerSpy)
+
+				depsTask := flow.Define(goyek.Task{
+					Name: "deps",
+					Action: func(a *goyek.A) {
+						a.Cleanup(onceCall(t, a.Name()+" cleanup"))
+						onceCall(t, a.Name())()
```

Comment:
This does not test anything apart from the `t.Cleanup` from `testing` package :wink: 
I do not think there is a possibility that this would ever fail (apart from a bug in `t.Cleanup`).